lib/mirage.mli

```diff
@@ -376,18 +379,20 @@ val direct_stackv4 :
 val socket_stackv4 :
   ?group:string -> Ipaddr.V4.t list -> stackv4 impl
 
+type arp_sig = ?clock:mclock impl -> ?time:time impl -> ethernet impl -> arpv4 impl
+
 (** Build a stackv4 by looking up configuration information via QubesDB,
  *  building an ipv4, then building a stack on top of that. *)
-val qubes_ipv4_stack : ?group:string -> ?qubesdb:qubesdb impl -> network impl -> stackv4 impl
+val qubes_ipv4_stack : ?group:string -> ?qubesdb:qubesdb impl -> ?arp:arp_sig -> network impl -> stackv4 impl
```

Comment:
this is related to https://github.com/mirage/mirage/pull/743#issuecomment-268390620 - I personally don't think there is much value in clock/time, but I'd want to use only a single time&clock in a unikernel.  if we provide `?clock` for some combinators, but use `default_clock` hardcoded in others, this does not allow to use a custom clock.

the other reason is that ocaml complained that the signatures don't match (I suspect I've to partially apply the functions with `clock` and `time` being `None`).